### DISCUSSION TITLE
Importing built-in fixtures

### DISCUSSION THREAD

**Discussion by pnuu** [pytest-dev/pytest#10648] [General] [answered]
`Explicit is better than implicit.` - PEP20, Zen of Python.

I'd really like to have all the used fixtures shown in the test module, either by defining them myself, or by showing where they're coming from by explicitly importing them.

So, what's the proper way to import the built-in fixtures? I could use `from _pytest.tmpdir import tmp_path`, but importing from a private package seems also wrong.

**Comment by nicoddemus** [comment, marked as answer]
There is no support for that currently, and no planned support either. See #3834 for a lengthy discussion, including alternatives/ideas.

**Reply by pnuu** [reply]
Thanks! I'll browse that through.

**Comment by pnuu** [comment]
To reduce the confusion for new developers who are not used to pytest fixtures, we are considering adding a note of at the top of our test modules listing which builtin/`conftest.py` fixtures are used. A more detailed explanation in our developer documentation should also help.
